tree/ntuple/v7/src/RNTupleSerialize.cxx

```diff
@@ -247,13 +247,21 @@ std::uint32_t SerializePhysicalColumn(const ROOT::Experimental::RColumnDescripto
 std::uint32_t SerializeColumnsOfFields(const ROOT::Experimental::RNTupleDescriptor &desc,
                                        std::span<const ROOT::Experimental::DescriptorId_t> fieldList,
                                        const ROOT::Experimental::Internal::RNTupleSerializer::RContext &context,
-                                       void *buffer)
+                                       void *buffer, bool forHeaderExtension)
 {
    auto base = reinterpret_cast<unsigned char *>(buffer);
    auto pos = base;
    void **where = (buffer == nullptr) ? &buffer : reinterpret_cast<void **>(&pos);
 
+   const auto *xHeader = !forHeaderExtension ? desc.GetHeaderExtension() : nullptr;
```

Comment:
`desc.GetHeaderExtension()` already returns `nullptr` if there is none. Is there any particular reason to explicitly use a boolean and set it to `nullptr` (same question for `SerializeAliasColumnsOfFields`)?

Reply:
The reason is that we can have a header extension even if `forHeaderExtension == false`, but in that case we want to ignore it